pkgs/db/pebble.go

```diff
@@ -0,0 +1,423 @@
+//go:build pebbledb
+
+package db
+
+import (
+	"bytes"
+	"fmt"
+	"path/filepath"
+
+	"github.com/cockroachdb/pebble"
+)
+
+func init() {
+	dbCreator := func(name string, dir string) (DB, error) {
+		return NewPebbleDB(name, dir)
+	}
+	registerDBCreator(PebbleDBBackend, dbCreator, false)
+}
+
+// PebbleDB is a PebbleDB backend.
+type PebbleDB struct {
+	db *pebble.DB
+}
+
+var _ DB = (*PebbleDB)(nil)
+
+func (db *PebbleDB) DB() *pebble.DB {
+	return db.db
+}
+
+// NB:  A lot of the code in this file is sourced from here: https://github.com/cockroachdb/pebble/blob/master/cmd/pebble/db.go
+// NB: This was my best working commit from today July 27 2022: https://github.com/notional-labs/tm-db/tree/6a41b774b9c362cac7d22156fa1021780d801f8a
+
+func NewPebbleDB(name string, dir string) (DB, error) {
+	dbPath := filepath.Join(dir, name+".db")
+	//	cache := pebble.NewCache(1024 * 1024 * 32)
+	//	defer cache.Unref()
+	opts := &pebble.Options{
+		//		Cache:                       cache,
+		//		FormatMajorVersion:          pebble.FormatNewest,
+		//		L0CompactionThreshold:       2,
+		//		L0StopWritesThreshold:       1000,
+		//		LBaseMaxBytes:               64 << 20, // 64 MB
+		//		Levels:                      make([]pebble.LevelOptions, 7),
+		//		MaxConcurrentCompactions:    3,
+		//		MaxOpenFiles:                1024,
+		//		MemTableSize:                64 << 20,
+		//		MemTableStopWritesThreshold: 4,
+	}
+	/*
+		for i := 0; i < len(opts.Levels); i++ {
+			l := &opts.Levels[i]
+			l.BlockSize = 32 << 10       // 32 KB
+			l.IndexBlockSize = 256 << 10 // 256 KB
+			l.FilterPolicy = bloom.FilterPolicy(10)
+			l.FilterType = pebble.TableFilter
+			if i > 0 {
+				l.TargetFileSize = opts.Levels[i-1].TargetFileSize * 2
+			}
+			l.EnsureDefaults()
+		}
+	*/
+	//	opts.Levels[6].FilterPolicy = nil
+	//	opts.FlushSplitBytes = opts.Levels[0].TargetFileSize
```

Comment:
Can you propose a first set of settings + a link to the official documentation so we can improve it later?